packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart

```diff
@@ -230,26 +242,51 @@ class WKPreferences {
 ///
 /// Wraps [WKWebsiteDataStore](https://developer.apple.com/documentation/webkit/wkwebsitedatastore?language=objc).
 class WKWebsiteDataStore {
-  /// Constructs a [WKWebsiteDataStore] that is owned by [configuration].
-  @visibleForTesting
-  WKWebsiteDataStore.fromWebViewConfiguration(
-    WKWebViewConfiguration configuration, {
+  WKWebsiteDataStore._({
     BinaryMessenger? binaryMessenger,
     InstanceManager? instanceManager,
   }) : _websiteDataStoreApi = WKWebsiteDataStoreHostApiImpl(
           binaryMessenger: binaryMessenger,
           instanceManager: instanceManager,
-        ) {
-    _websiteDataStoreApi.createFromWebViewConfigurationForInstances(
-      this,
+        );
+
+  factory WKWebsiteDataStore._defaultDataStore() {
+    throw UnimplementedError();
+  }
+
+  /// Constructs a [WKWebsiteDataStore] that is owned by [configuration].
+  @visibleForTesting
+  factory WKWebsiteDataStore.fromWebViewConfiguration(
+    WKWebViewConfiguration configuration, {
+    BinaryMessenger? binaryMessenger,
+    InstanceManager? instanceManager,
+  }) {
+    final WKWebsiteDataStore websiteDataStore = WKWebsiteDataStore._(
+      binaryMessenger: binaryMessenger,
+      instanceManager: instanceManager,
+    );
+    websiteDataStore._websiteDataStoreApi
+        .createFromWebViewConfigurationForInstances(
+      websiteDataStore,
       configuration,
     );
+    return websiteDataStore;
   }
 
+  /// Default data store that stores data persistently to disk.
+  static final WKWebsiteDataStore defaultDataStore =
+      WKWebsiteDataStore._defaultDataStore();
+
   final WKWebsiteDataStoreHostApiImpl _websiteDataStoreApi;
 
+  /// Manages the HTTP cookies associated with a particular web view.
+  late final WKHttpCookieStore httpCookieStore =
+      WKHttpCookieStore.fromWebsiteDataStore(this);
+
   /// Removes website data that changed after the specified date.
-  Future<void> removeDataOfTypes(
+  ///
+  /// Returns whether any data was removed.
+  Future<bool> removeDataOfTypes(
```

Comment:
This Objective-C class doesn't return a `bool`, but I decided to just add this feature for simplicity.